tests/swap_test.py

```diff
@@ -109,6 +109,32 @@ def test_swapon_pagesize(self):
         with self.assertRaises(BlockDev.SwapPagesizeError):
             BlockDev.swap.swapon(self.loop_dev)
 
+    def _remove_map(self, map_name):
+        run("dmsetup remove -f %s" % map_name)
+
+    def test_swapstatus_dm(self):
+        """Verify that swapstatus works correctly with DM devices"""
+
+        dm_name = "swapstatus-test"
+
+        self.addCleanup(self._remove_map, dm_name)
+        run("dmsetup create %s --table \"0 $((`lsblk -osize -b %s -nd`/512)) linear %s 0\"" % (dm_name, self.loop_dev, self.loop_dev))
+
+        succ = BlockDev.swap_mkswap("/dev/mapper/%s" % dm_name, None, None)
+        self.assertTrue(succ)
+
+        on = BlockDev.swap_swapstatus("/dev/mapper/%s" % dm_name)
+        self.assertFalse(on)
+
+        succ = BlockDev.swap_swapon("/dev/mapper/%s" % dm_name, -1)
+        self.assertTrue(succ)
+
+        on = BlockDev.swap_swapstatus("/dev/mapper/%s" % dm_name)
+        self.assertTrue(on)
+
+        succ = BlockDev.swap_swapoff("/dev/mapper/%s" % dm_name)
+        self.assertTrue(succ)
+
```

Comment:
One more `swapstatus()` check here would be nice.

Reply:
Fixed.